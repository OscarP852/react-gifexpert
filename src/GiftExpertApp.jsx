import { useState } from 'react';
import { AddCategory, GifGrid } from './components';
export const GiftExpertApp = () => {

    const [categories, setCategories] = useState([]);

    const onAddCategory = ( newCategorie ) =>{

        if(categories.includes(newCategorie)) return;
        setCategories([newCategorie, ...categories]);

    }
    
    return (
        <>
            <h1>Buscador de Imagenes</h1>
            <AddCategory 
                onNewCategory = { onAddCategory }
            />
                { 
                    categories.map( category => (
                    <GifGrid 
                        key={ category }
                        category={ category }
                    />
                    ))
                }
        </>
    );
}