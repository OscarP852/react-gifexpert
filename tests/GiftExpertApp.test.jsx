import { render, screen, fireEvent } from '@testing-library/react';
import { GiftExpertApp } from '../src/GiftExpertApp'

describe('Pruebas en <GiftExpertApp/>',()=>{

    test('evaluar el snapshot', () => {
        const { container } = render( <GiftExpertApp/>);
        expect( container ).toMatchSnapshot();
    });

    test('Debe validar que las categorias no se repitan', () => {
        render( <GiftExpertApp />);
 
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
       
        fireEvent.input( input, { target: { value: 'Dir En Grey' } } );
        fireEvent.submit( form );
        fireEvent.input( input, { target: { value: 'Dir En Grey' } } );
        fireEvent.submit( form );
        
        expect( screen.getByText('Dir En Grey') ).toBeTruthy();
    });
    test('no debe añadir una categoría existente', () => {
 
        render( <GiftExpertApp />);
 
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
        
        fireEvent.input( input, { target: { value: 'Dir En Grey' } } );
        fireEvent.submit( form );
        
        expect( screen.getAllByText('Dir En Grey').length ).toBe( 1 );
 
    });
 
});