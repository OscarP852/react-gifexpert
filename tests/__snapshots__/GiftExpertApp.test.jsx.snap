// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExpertApp/> evaluar el snapshot 1`] = `
<div>
  <h1>
    Buscador de Imagenes
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Imagenes"
      type="text"
      value=""
    />
  </form>
</div>
`;
